modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/OpenAPIV3Parser.java

```diff
@@ -27,12 +27,18 @@
     public SwaggerParseResult readLocation(String url, List<AuthorizationValue> auth, ParseOptions options) {
         SwaggerParseResult result = new SwaggerParseResult();
         try {
-            // TODO
             JsonNode node = YAML_MAPPER.readValue(new URL(url), JsonNode.class);
             if(node != null && node.get("openapi") != null) {
                 JsonNode version = node.get("openapi");
+                if(auth == null) {
+                    auth = new ArrayList<>();
+                }
                 if(version.asText() != null && version.asText().startsWith("3.0")) {
-                    result = new OpenAPIDeserializer().deserialize(node);
+                    if (options != null) {
+                        if (options.isResolve()) {
+                            result.setOpenAPI(new OpenAPIResolver(new OpenAPIDeserializer().deserialize(node).getOpenAPI(), auth, null).resolve());
+                        }
+                    }
                 }
```

Comment:
I think you're missing the `else` here.  If you don't resolve, you can still deserialize and set it in the object.